iris_grib/grib_phenom_translation.py

```diff
@@ -331,3 +332,59 @@ def cf_phenom_to_grib2_info(standard_name, long_name=None):
     if standard_name is not None:
         long_name = None
     return _CF_GRIB2_TABLE[(standard_name, long_name)]
+
+
+class GribCode(namedtuple('GribCode',
+                          'edition discipline category number')):
+    """
+    An object representing a specific Grib phenomenon identity.
+
+    Basically a namedtuple of (edition, discipline, category, number).
+
+    Also provides a string representation, and supports creation from: another
+    similar object; a tuple of numbers; or any string with 4 separate decimal
+    numbers in it.
+
+    """
+    __slots__ = ()
+
+    def __new__(cls, edition_or_string,
+                discipline=None, category=None, number=None):
+        if (all(arg is None
+                for arg in (discipline, category, number))):
+            # Only 1 arg : can  be existing object, or a string.
+            # Convert to a string : also covers case of a GribCode to copy.
```

Comment:
Note: a tuple of numbers will also work here via the str() route.
Worth adding to comment